### ISSUE TITLE
Custom Rockon management deletion script enhancement.

### ISSUE THREAD

**Issue by Hooverdan96** [rockstor/rockstor-core#2549]
This issue is triggered by the experience of forum user [`aymenso`](https://forum.rockstor.com/u/aymenSo):
https://forum.rockstor.com/t/get-returned-more-than-one-rockon-it-returned-2/8790

the Rockon delete script should consider an inconsistency scenario, where the same rockon is represented twice in the database: this should not occur, however we should probably prepare for that case, while working towards avoiding this scenario (possibly occurring during the json ingestion step).

https://github.com/rockstor/rockstor-core/blob/0249f5283203461d71093447611c762901f6f378/src/rockstor/scripts/rockon_delete.py#LL40C1-L57C1

May be the above scenario can be addressed by adding an outer loop over the `ro` object, so it would treat possible duplicates as individual rockons and delete accordingly?

Additional observation
While trying to reproduce the inconsistency (which I was not successful in), this is what I saw (all on 4.5.9-1):

- taking example of scrutiny rockon. Create two custom json files, differentiated by file name (scrutiny1.json, scrutiny2.json) and `containers` name (scrutiny1 and scrutiny2), but having the same Rockon name: `Scrutiny.local`. I made a non-relevant (I think) change to the description to be able to differentiate them, everything else remained the same.

- After a refresh, only one Rockon appeared, namely the scrutiny2.json's content.

<img width="886" alt="image" src="https://user-images.githubusercontent.com/35113775/236507338-5e8d0a1e-84b5-46c9-83e3-402d19337cae.png">

- Without installing it, but directly going for the script for rockon deletion, it removed the custom scrutiny entry without error.

- Redoing the setup. Now installing the visible custom Rockon - no issue
<img width="1425" alt="image" src="https://user-images.githubusercontent.com/35113775/236508707-383677cd-907f-4aae-84d8-2aa80ba007fd.png">

- Refreshing the Rockon list, while the Scrutiny.local is installed. scrutiny1.json contents are not picked up, i.e. there is still only 1 Scrutiny.local (installed).

- Stopping and uninstalling Scrutiny.local using the UI. No issue. Selecting refresh, still only 1 (scrutiny2.json) is visible in the list.

- Reinstalling Scrutiny.local and while it's running use the delete script. Reload the web page, Scrutiny.local rockon has disappeared completely (as expected). Now Refresh Rockons again. scrutiny2.json is picked up as before.

So, not sure really how this inconsistency can happen

**Comment by FroggyFlox** [comment]
I'm currently working on this.

**Comment by FroggyFlox** [comment]
I now have a working fix; following @Hooverdan96's example of a `Scrutiny.local` Rock-On, with its entry in `storageadmin.rockon` manually duplicated.
We have:
```
rockdev:/opt/rockstor # poetry run delete-rockon "Scrutiny.local"
The metadata for the Rock-On named Scrutiny.local (id: 81) has been deleted from the database
The metadata for the Rock-On named Scrutiny.local (id: 80) has been deleted from the database
```

All entries from these two Rock-Ons were confirmed to be removed from the database.

**Comment by Hooverdan96** [comment]
based on the forum response, sounds like the changed script is working! Excellent.
In your changes, does it make sense somewhere in the usage reference (and subsequently in the rockstor-doc) to mention that the Rockon name is case-sensitive (since you're already making changes in the code)?

**Comment by FroggyFlox** [comment]
Excellent point and idea, @Hooverdan96 !
I'll include that in a couple spots in the rockon-delete script.

**Comment by Hooverdan96** [comment]
Yes, just like that!!! Thanks.

**Comment by phillxnet** [comment]
Closing as:
Fixed by #2552
